src/sql/workbench/services/fileBrowser/browser/urlBrowserDialog.ts

```diff
@@ -0,0 +1,477 @@
+				.then(getBlobContainersResult => this.setBlobContainersSelectorBoxOptions(getBlobContainersResult.blobContainers))
+				.catch(getBlobContainersResult => this.setBlobContainersSelectorBoxErrors(getBlobContainersResult.errors));
+		} else {
+			this._selectedStorageAccount = '';
+			this.setBlobContainersSelectorBoxErrors({});
+		}
+	}
+
+	private setBlobContainersSelectorBoxOptions(blobContainers: BlobContainer[]) {
+		this._blobContainers = blobContainers;
+		const blobContainersDisplayNames: string[] = this._blobContainers.map(blobContainer => blobContainer.name);
+		this._blobContainerSelectorBox.setOptions(blobContainersDisplayNames);
+		this._blobContainerSelectorBox.select(0);
+		if (this._blobContainers.length === 0) {
+			this._blobContainerSelectorBox.disable();
+		} else {
+			this._blobContainerSelectorBox.enable();
+		}
+	}
+
+	private setBlobContainersSelectorBoxErrors(errors: any) {
+		this._blobContainers = [];
+		this._blobContainerSelectorBox.setOptions([]);
+		this._blobContainerSelectorBox.select(0);
+		this._blobContainerSelectorBox.disable();
+	}
+
+	private onBlobContainersSelectorBoxChanged(checkedBlobContainer: number) {
+		this._sasInputBox.value = '';
+		if (this._restoreDialog) {
+			if (this._blobContainers.length !== 0) {
+				this._selectedBlobContainer = this._blobContainers[checkedBlobContainer];
+				this._azureAccountService.getBlobs(this._selectedAccount, this._selectedSubscription, this._selectedStorageAccount, this._selectedBlobContainer.name, true)
+					.then(getBlobsResult => this.setBackupFilesOptions(getBlobsResult))
+					.catch(getBlobsResult => this.setBackupFilesSelectorError(getBlobsResult));
+			} else {
+				this._selectedBlobContainer = '';
+				this.setBackupFilesSelectorError({});
+			}
+		}
+		this.enableCreateCredentialsButton();
+	}
+
+	private setBackupFilesOptions(getBlobsResult: GetBlobsResult) {
+		this._backupFiles = getBlobsResult.blobs;
+		const backupFilesDisplayNames: string[] = this._backupFiles.map(backupFile => backupFile.name);
+		this._backupFileSelectorBox.setOptions(backupFilesDisplayNames);
+		this._backupFileSelectorBox.select(0);
+		if (this._backupFiles.length === 0) {
+			this._backupFileSelectorBox.disable();
+		} else {
+			this._backupFileSelectorBox.enable();
+		}
+	}
+
+	private setBackupFilesSelectorError(errors: any) {
+		this._backupFileSelectorBox.setOptions([]);
+		this._backupFileSelectorBox.select(0);
+		this._backupFileSelectorBox.disable();
+	}
+
+	private setBackupFileDefaultValue() {
+		this._backupFileInputBox.value = this._defaultBackupName;
+	}
+
+	public open(ownerUri: string,
+		expandPath: string,
+		fileFilters: [{ label: string, filters: string[] }],
+		fileValidationServiceType: string,
+	): void {
+		this._ownerUri = ownerUri;
+		this.enableOkButton();
+		this.enableCreateCredentialsButton();
+		this.spinner = true;
+		this.show();
```

Comment:
Is this really going to take long enough that it's worth having the spinner show?